### ISSUE TITLE
[ENH] `make_forecasting_scorer` should expose additional parameters as `__init__` args

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#4834]
`make_forecasting_scorer` currently does not convert extra parameters to parameters of the dynamic forecasting metric - but it would be nice if it did.

Related issue is the bug https://github.com/sktime/sktime/issues/4827 where extra parameters (`squared` in the example) caused the metric to crash, the fix now ignores them.

But, in the optimal case, the extra parameters should be `__init__` parameters of the resulting metric. This will probably require some inspection or meta-programming - good first issue for anyone who knows how to create a dynamic `__init__` signature.

**Comment by yarnabrina** [comment]
Will you allow `*args` and `**kwargs`?

**Comment by fkiraly** [comment]
No, this is actually part of the `scikit-learn` compatibility contract afaik - `__init__` must not have `args` or `kwargs`, since the information of which parameters there are is obtained from inspecting the `__init__` signature, rather than from a dedicated method.

Whether one likes this design decision or not, it is what is required for `sklearn` compatibility...

**Comment by yarnabrina** [comment]
I was not aware of this restriction by `sklearn`, but `make_scorer` does accept arbitrary keyword arguments to pass to the scoring function.

See the docs [here](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.make_scorer.html#sklearn.metrics.make_scorer), or the in the source code:

https://github.com/scikit-learn/scikit-learn/blob/7f9bad99d/sklearn/metrics/_scorer.py#L644-L650

**Comment by fkiraly** [comment]
> but `make_scorer` does accept arbitrary keyword arguments to pass to the scoring function.

Yes, but that afaik substitutes them, rather than exposing them as parameters.
